pokemon.go

```diff
@@ -79,10 +79,20 @@ func (p Pokemon) String() string {
 		p.Gender, p.Level, p.CurrentHP, p.Stats[STAT_HP])
 }
 
+// Applies damage to a Pokemon. May cause Pokemon to faint.
+func (p *Pokemon) damage(amount uint) {
+	if p.CurrentHP >= amount {
+		p.CurrentHP -= amount
+	} else {
+		// prevent underflow
+		p.CurrentHP = 0
+	}
+}
+
 // Restore HP to a Pokemon. Can also be used to revive a fainted Pokemon.
-func (p *Pokemon) RestoreHP(amount uint) Transaction {
+func (p *Pokemon) heal(amount uint) {
 	if diff := p.Stats[STAT_HP] - p.CurrentHP; diff <= amount {
 		amount = diff
 	}
-	return HealTransaction{Target: p, Amount: amount}
+	p.CurrentHP += amount
 }
```

Comment:
Again, this logic is extremely trivial, and these functions are only used in one spot. All changes to battle state have to go through transactions anyway, so we should just test the whole transaction. I'd prefer if these functions didn't exist.

Reply:
Remove these, as indicated in previous review.